telegram/bot/bot.go

```diff
@@ -0,0 +1,112 @@
+package bot
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/moira-alert/notifier"
+	logging "github.com/op/go-logging"
+	"github.com/tucnak/telebot"
+)
+
+type bot struct {
+	key      string
+	telebot  *telebot.Bot
+	messages chan telebot.Message
+	db       notifier.Database
+}
+
+// Sender sends message
+type Sender interface {
+	Send(login, message string) error
+}
+
+// Starter starts bot
+type Starter interface {
+	Start() error
+}
+
+// Bot implements bot
+type Bot interface {
+	Sender
+	Starter
+}
+
+var (
+	logger *logging.Logger
+)
+
+// StartBot start a bot
+func StartBot(key string, log *logging.Logger, db notifier.Database) Bot {
+	logger = log
+	messages := make(chan telebot.Message)
+
+	api := &bot{
+		key:      key,
+		db:       db,
+		messages: messages,
+	}
+	go api.Start()
+
+	return api
+}
+
+func (b *bot) Start() error {
+	var err error
+	b.telebot, err = telebot.NewBot(b.key)
+	if !b.db.NotifierRegistered() {
```

Comment:
I see a race condition here.

1. notifier1 checks registration, it is unregistered.
2. notifier2 checks registration, it is unregistered.
3. notifier1 registers and starts bot.
4. notifier2 registers and starts bot.